components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/ActionManagementServiceImpl.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.action.management;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.action.management.constant.ActionMgtConstants;
+import org.wso2.carbon.identity.action.management.dao.impl.ActionManagementDAOImpl;
+import org.wso2.carbon.identity.action.management.dao.impl.CacheBackedActionMgtDAO;
+import org.wso2.carbon.identity.action.management.exception.ActionMgtClientException;
+import org.wso2.carbon.identity.action.management.exception.ActionMgtException;
+import org.wso2.carbon.identity.action.management.model.Action;
+import org.wso2.carbon.identity.action.management.util.ActionManagementUtil;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * Action management service.
+ */
+public class ActionManagementServiceImpl implements ActionManagementService {
+
+    private static final Log LOG = LogFactory.getLog(ActionManagementServiceImpl.class);
+    private static final ActionManagementService INSTANCE = new ActionManagementServiceImpl();
+    private static final CacheBackedActionMgtDAO CACHE_BACKED_DAO =
+            new CacheBackedActionMgtDAO(new ActionManagementDAOImpl());
+
+    private ActionManagementServiceImpl() {
+    }
+
+    public static ActionManagementService getInstance() {
+
+        return INSTANCE;
+    }
+
+    @Override
+    public Action addAction(String actionType, Action action, String tenantDomain) throws ActionMgtException {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Adding Action for Action Type: " + actionType);
+        }
+        String resolvedActionType = getActionTypeFromPath(actionType);
+        String generatedActionId = UUID.randomUUID().toString();
+        // Check whether the maximum allowed actions per type is reached.
+        validateMaxActionsPerType(resolvedActionType, tenantDomain);
```

Comment:
move this statement before generating the id

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/5771/commits/a96c6e8c2f905af6c7fd842a69a4d972997473a1